### ISSUE TITLE
AWSSQS - Message is Acknowledged and deleted after Requeue

### ISSUE THREAD

**Issue by honkuan86** [BrighterCommand/Brighter#1668]
### Describe the bug
There is an issue with the MessagePump that will Acknowledge messages after requeue. While it may work for other MessageGateway, but for AwsSqs it behave unexpectedly, because in AwsSqs, Acknowledge a message means deleting off from the queue. So the requeued message will be deleted before reach to DLQ.

Noticed the behavior changed after this commit #1544 

A indication or a flag is needed for a requeued message, so that it will not be deleted when the message is being Acknowledged.

**Comment by honkuan86** [comment]
Hi @iancooper , @preardon , is this behavior expected? What I'm trying to achieve is that, when a handler throws a DerferMessageAction, I'm expecting the message to be pushed back to the queue until the max receive is reached then Sqs will move that message to DLQ. But it is not working as expected currently. Am I understand it wrongly ?

**Comment by preardon** [comment]
@honkuan86 the expected behaviour is that it ACK's the message and puts in back on the back of the queue until the max retry count has been reached, then it will put it on the DLQ

**Comment by iancooper** [comment]
Yeah, that could be an issue, as @preardon  suggests we would expect if you ask for a requeue we would push back onto the queue, not delete. So the place to look will be at how we have implemented requeue, where we might have just called ack, because we do that in some other transports
